api/models/post/util.js

```diff
@@ -192,14 +207,22 @@ export const createPost = (userId, params) =>
       {children: params.requests, transacting: trx}
     )))))
 
-export const createThread = (userId, params) =>
-  setupNewPostAttrs(userId, params)
+export const findOrCreateThread = (userId, participantIds) =>
+  Post.query(q => {
+    q.join('follows', 'follows.post_id', 'posts.id')
```

Comment:
i don't think you need to join here, because you don't reference `follows` explicitly anywhere else -- the `Follow.query()` lines create subqueries